### ISSUE TITLE
S3 write timeout (Ruby 2.0)

### ISSUE THREAD

**Issue by ampedandwired** [aws/aws-sdk-ruby#241]
This is an odd one. The following code generates a timeout for me when run with ruby 2.0.0-p0 but it's OK with 1.9.3-p385. Using aws-sdk 1.9.2.

```ruby
require 'aws'
AWS.config(access_key_id: 'KEY', secret_access_key: 'SECRET')
s3 = AWS::S3.new
emr = AWS::EMR.new
bucket_name = 'some-bucket'
s3.buckets[bucket_name].exists?
emr.job_flows.each { }
puts "Uploading file"
s3.buckets[bucket_name].objects['foo.sh'].write(file: '/path/to/file.txt')
```

The calls to `bucket.exists?` and `job_flows.each` are significant - if you remove either of these the write succeeds.

The error generated:
```
Uploading file
/home/charles/.rbenv/versions/2.0.0-p0/gemsets/awstest/gems/aws-sdk-1.9.2/lib/aws/core/client.rb:360:in `return_or_raise': Your socket connection to the server was not read from or written to within the timeout period. Idle connections will be closed. (AWS::S3::Errors::RequestTimeout)
	from /home/charles/.rbenv/versions/2.0.0-p0/gemsets/awstest/gems/aws-sdk-1.9.2/lib/aws/core/client.rb:461:in `client_request'
	from (eval):3:in `put_object'
	from /home/charles/.rbenv/versions/2.0.0-p0/gemsets/awstest/gems/aws-sdk-1.9.2/lib/aws/s3/s3_object.rb:1669:in `write_with_put_object'
	from /home/charles/.rbenv/versions/2.0.0-p0/gemsets/awstest/gems/aws-sdk-1.9.2/lib/aws/s3/s3_object.rb:608:in `write'
	from awstest.rb:15:in `<main>'
```

**Comment by uberllama** [comment]
Thanks for your hard work over such a long period of time on this! Very exciting.

**Comment by gonzedge** [comment]
👏

**Comment by trevorrowe** [comment]
I should add a caveat that the fix only applies for Ruby versions 1.9.3+. Ruby 1.8.7 and Ruby 1.9.2 do not support the Expect 100-continue and therefore will continue to have their connections closed by Amazon S3 under certain circumstances.

**Comment by justincase** [comment]
[Ruby 1.8.7 is retired](https://www.ruby-lang.org/en/news/2013/06/30/we-retire-1-8-7/) and [1.9.3 will be EOL soon](Support for Ruby version 1.9.3 will end on February 23, 2015).

:fireworks: :ship: :+1:
